configs/components/facter-ng.rb

```diff
@@ -22,7 +22,7 @@
   # Remove required_ruby_version from gemspec, this is needed for cross-compiled
   # platforms as ruby version is lower than 2.3 on these platforms.
   if platform.is_cross_compiled?
-    sed_pattern = %(s/spec.required_ruby_version = '~> 2.3'/ /)
+    sed_pattern = %(s/spec.required_ruby_version = '>= 2.3', '< 4.0'/ /)
```

Comment:
Can we just match `spec.required_ruby_version` and remove the entire line without matching the specific version? I'm not sure how to do that off the top of my head with sed but it might be a more future-proof fix.